### PR TITLE
Rename Java IDE to idelab

### PR DESCRIPTION
**Pull request by molly-moen**
Renaming Java IDE to idelab to protect against changing from Java to some other language in the future. The defining feature of the lab is the IDE, not Java.

This PR only contains the changes required for name changes, any "new files" are just renames. The IDE will now be available for admins and levelbuilders at `studio.code.org/idelab`.

## Testing story
Tested locally.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/idelab/Idelab.jsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import React from 'react';
```
Comment (jmkulwik) [inline apps/src/idelab/Idelab.jsx:1]: Should we review the name with Star Labs? (cc: @moneppo)

File: dashboard/config/routes.rb [hunk 2]
```diff
@@ -774,7 +774,7 @@
 
   post '/i18n/track_string_usage', action: :track_string_usage, controller: :i18n
 
-  get 'java_ide', to: 'java_ide#index'
+  get 'idelab', to: 'idelab#index'
```
Comment (jmkulwik) [inline dashboard/config/routes.rb:777]: Do we still want the url to say "java_ide?" (cc: @moneppo)

Comment (jmkulwik) [reply dashboard/config/routes.rb:777]: i.e. then in the future we might also have a route for "python_ide"

Comment (molly-moen) [reply dashboard/config/routes.rb:777]: fwiw I think we can get rid of this url completely once we are integrated with code studio

Comment (jmkulwik) [reply dashboard/config/routes.rb:777]: 👍

**Review by jmkulwik — APPROVED** [review]
This looks good to me. It'd be nice to get a quick 👍 from Star Labs on the name - if only so they know what the new directory is for.

**Pull request merged**